prog/weaver/discovery.go

```diff
@@ -0,0 +1,103 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/gorilla/mux"
+
+	"github.com/weaveworks/weave/ipam"
+)
+
+// TODO: move these definitions somewhere more shareable
```

Comment:
*Minor*: ICYMI, you may want to deal with this `TODO` out before we merge the PR?